snapmaker/src/module/toolhead_laser.cpp

```diff
@@ -1025,16 +1025,34 @@ void ToolHeadLaser::InlineDisable() {
 
 void ToolHeadLaser::SetOutputInline(float power) {
   CheckFan(power);
-  planner.laser_inline.status.isEnabled = true;
 
   SetPower(power);
   planner.laser_inline.power = power_pwm_;
+  
+  if (power_pwm_ > 0)
+    planner.laser_inline.status.isEnabled = true;
+  else
+    planner.laser_inline.status.isEnabled = false;
+
+  if (laser->device_id_ == MODULE_DEVICE_ID_10W_LASER && power_pwm_ > 0) {
+    if (laser_10w_status_ == LASER_10W_DISABLE) {
+      // Send the enable command only when the power is disabled
+      LaserControl(1);
```

Comment:
Maybe not part of this commit, but should `LaserControl` take in a boolean? I understand that you kinda want to avoid converting the value before sending it over the CAN bus, but it's being coerced to a boolean in `LaserControl` and it's not exactly straightforward when the method takes in an int. Either that, or the setting of `laser_10w_status_` should be removed from `LaserControl` such that it's not sending stuff over the can bus _and_ setting some internal state to the controller.